Classes/Fluid/ViewHelper/ComponentRenderer.php

```diff
@@ -279,6 +288,72 @@ public function prepareArguments()
         return $this->argumentDefinitions;
     }
 
+    /**
+     * Prepares a specified JSON schema for usage
+     *
+     * @param mixed $schema
+     * @return Schema
+     */
+    public function prepareSchema($schema)
+    {
+        $schemaFile = null;
+        if (is_array($schema) || is_object($schema)) {
+            // Normalize arrays and objects that come directly from fluid templates
+            $schema = json_encode($schema);
+            $hash = sha1($schema);
+        } else {
+            $schema = trim($schema);
+            if ($schema{0} === '{') {
```

Comment:
That seems to be a quite weak check. Maybe something like the following would suit your needs better:
```
json_decode($schema);
    if (json_last_error() === JSON_ERROR_NONE) {
```